### PR TITLE
ns:number function does not operate on a node #15

### PR DESCRIPTION
**Pull request by wjgerritsen-0001**

### CODE REVIEW

File: tests/XPath2.Tests/XPath2Tests.cs [hunk 1]
```diff
@@ -105,5 +105,23 @@ public void XPath2_substring()
...
         }
+
+        [Fact]
+        public void XPath2_number()
+        {
+            var xml = new XmlDocument
+            {
+                InnerXml = @"
+                    <root num='134'>
+                        <numeric>456</numeric>
```
Comment (StefH) [inline tests/XPath2.Tests/XPath2Tests.cs]: Use a `.` here for testing purposes.
```suggestion
                        <numeric>456.78</numeric>
```

File: src/XPath2/CoreFuncs.cs [hunk 2]
```diff
@@ -1394,12 +1394,13 @@ public static double Number(XPath2Context context, IContextProvider provider)
 
         public static double Number(XPath2Context context, object value)
         {
-            if (value == Undefined.Value || !(value is IConvertible))
+            string stringValue = StringValue(context, value);
+            if (string.IsNullOrWhiteSpace(stringValue))
                 return double.NaN;
```
Comment (StefH) [inline src/XPath2/CoreFuncs.cs]: Add { and }

```suggestion
{
    return double.NaN;
}
```

File: tests/XPath2.Tests/XPath2Tests.cs (hunk 1 above)
Comment (wjgerritsen-0001) [reply tests/XPath2.Tests/XPath2Tests.cs]: I will update the test

**Comment by wjgerritsen-0001** [comment]
With latest changes, the test suite now has 12952 from 15133 (85.59%) conformance testcases succeeded.

**Pull request merged**